template/lib.typ

```diff
@@ -227,27 +227,151 @@ text)
+          title_left = strong([#numbering(numeration,page)] + h(1.4em) + chapter + title.body);
+        }
+
+        // Right headers for the main content
+        if locate(loc).page() <= here().page() and title.level == 2 {
+          title_right = strong([#title_numbering ]+ title.body + h(1.4em) + [#numbering(numeration,page)]);
+        }
+
+        // Skip the first header of a section
+        if(title.level == 1 and locate(loc).page() == here().page()){
+          title_left = none;
+        }
+    }
+
+    // Header placement
+    if title_left != none{
+    if calc.odd(page){
+
+    set text(1em)  
+    place(top + right, dx: + 2em, dy: 3.5em,[#title_right] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(0pt,-10pt), end:(40em + 25pt,-10pt))
+
+    } else {
+   
+    set text(1em)  
+    place(top + left, dx: -2em, dy: 3.5em,[#title_left] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(-25pt,-10pt), end:(40em,-10pt))
+  
+    }
+    } else {
+
+      // If there is no header, place the page numbering in the middle
+      place(top + center , dx: -.025em, dy: 72.028em,[#numbering(numeration,page)])
+    }
+   }
+
+  } else {
+
+       context {
+    // Header for the others type of content    
+    let titles = query(selector(heading.where(level: 1)))
+
+    let title_left
+    let num_left
+    let title_right
+    let num_right
+
+    let page = counter(page).get().first() 
+
+    for title in titles {
+    let loc = title.location() 
+    let title_numbering = if title.numbering == none {
+          none
+        } else {
+          context numbering("1.", ..counter(heading).at(loc))
+        } 
+
+        // Prepare both left and right headers parts
+        if locate(loc).page() <= here().page()  {
+          num_left = strong([#numbering(numeration,page)]);
+          title_left = strong(title.body);
+          num_right = strong([#numbering(numeration,page)]);
+          title_right = strong(title.body);
+        }
+
+        // Skip the first header of a section
+        if(locate(loc).page() == here().page()){
+          title_left = none;
+        }
+    }
+
+    // Header placement
+    if title_left != none{
+    if calc.odd(page){
+
+    set text(1em)  
+    place(top + right, dx: + 2em, dy: 3.5em,[#num_right])
+    place(top + center, dx: 0em, dy: 3.5em,[#title_right])
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(0pt,-10pt), end:(40em + 25pt,-10pt))
+
+    } else {
+   
+    set text(1em)  
+    place(top + left, dx: -2em, dy: 3.5em,[#num_left])
+    place(top + center, dx: + 0em, dy: 3.5em,[#title_left] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(-25pt,-10pt), end:(40em,-10pt))
+  
+    }
+    } else {
+      // If there is no header, place the page numbering in the middle
+      place(top + center , dx: -.025em, dy: 72.028em,[#numbering(numeration,page)])
+    }
+   }
+   }
```

Comment:
Please be careful with indentation with blocks. Try that blocks always have their content indented two spaces more. Small detail, but improves readability. I remember there was a VSCode extension for formatting Typst code, remind me next week and I'll tell you which one it is

Reply:
could you please explain me what do you mean? We can talk tomorrow in the break :)